pkg/lib/infra/db/pool.go

```diff
@@ -1,25 +1,52 @@
 package db
 
 import (
+	"context"
 	"errors"
 	"sync"
 
 	"github.com/jmoiron/sqlx"
 )
 
+type PoolDB struct {
+	DB       *sqlx.DB
+	stmtLock *sync.RWMutex
+	stmts    map[string]*sqlx.Stmt
+}
```

Comment:
Implement Close() for PoolDB. It should Close all cached prepared statements, and then close the DB.